### PR TITLE
Fix syntax for enum classes and overridden members.

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/TypeSpec.kt [hunk 1]
```diff
@@ -233,31 +227,42 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
...
 
     ANNOTATION(
+        "annotation class",
         emptySet(),
-        setOf(KModifier.PUBLIC, KModifier.ABSTRACT))
+        setOf(KModifier.PUBLIC, KModifier.ABSTRACT));
+
+    override fun toString(): String {
+      return super.toString()
+    }
```
Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/TypeSpec.kt]: Delete

File: src/main/java/com/squareup/kotlinpoet/FunSpec.kt [hunk 2]
```diff
@@ -437,7 +437,7 @@ class FunSpec private constructor(builder: Builder) {
       val methodName = method.simpleName.toString()
       val funBuilder = FunSpec.builder(methodName)
 
-      funBuilder.addAnnotation(Override::class)
+      funBuilder.addModifiers(KModifier.OVERRIDE)
```
Comment (ronshapiro) [inline src/main/java/com/squareup/kotlinpoet/FunSpec.kt]: Can you fix the javadoc above?
```diff
-An [Override] annotation will be added.
+An `override` modifier will be added.
```

Comment (swankjesse) [inline src/main/java/com/squareup/kotlinpoet/FunSpec.kt]: Yeah, good catch. Fixed.

File: src/main/java/com/squareup/kotlinpoet/TypeSpec.kt (hunk 1 above)
Comment (swankjesse) [inline src/main/java/com/squareup/kotlinpoet/TypeSpec.kt]: Fixed!

**Pull request merged**